src/prepare.js

```diff
@@ -43,14 +45,19 @@ function createCompositeElementInstance(
 }
 
 function renderCompositeElementInstance(instance, context = {}) {
-  const childContext = Object.assign(
-    {},
-    context,
-    instance.getChildContext ? instance.getChildContext() : {},
-  );
+  const childContext = {
+    ...context,
+    ...(instance.getChildContext ? instance.getChildContext() : {}),
+  };
   return [instance.render(), childContext];
 }
 
+function renderFunctionElementInstance({ type, props }, context = {}) {
+  setDispatcherContext(context);
+  registerDispatcher();
```

Comment:
Is there a reason you are calling `registerDispatcher` every time?
If not, I would imagine it's best to set it once in `prepare` and only modify the context when needed. Otherwise it's quite confusing (and unnecessary) to call it on every function component invocation.

Reply:
No, there is no need to do it each time, I can wrap it in an `if`-statement if you want. I think it makes sense to do it in `renderFunctionElementInstance` rather than `prepare` though, as it's not used for any other elements.

Reply:
:+1:. My thought is that it makes sense to keep all "set up" at the start of prepare, even though you might not need it based on the components being rendered, but still it keeps everything in the same, logical place.

Reply:
I guess that makes some sense, I moved it to the beginning of `prepare`. I also removed the `renderFunctionElementInstance` function, as it was very simple and unnecessary after the React.Memo recursion thing.